src/chainstate/burn/operations/mod.rs

```diff
@@ -61,86 +106,151 @@ impl error::Error for Error {
         match *self {
             Error::ParseError => "Failed to parse transaction into Blockstack operation",
             Error::InvalidInput => "Invalid input",
-            Error::DBError(ref e) => e.description()
+
+            Error::BlockCommitPredatesGenesis => "Block commit predates genesis block",
+            Error::BlockCommitAlreadyExists => "Block commit commits to an already-seen block",
+            Error::BlockCommitNoLeaderKey => "Block commit has no matching register key",
+            Error::BlockCommitLeaderKeyAlreadyUsed => "Block commit register key already used",
+            Error::BlockCommitNoParent => "Block commit parent does not exist",
+            Error::BlockCommitBadInput => "Block commit tx input does not match register key tx output",
+
+            Error::LeaderKeyAlreadyRegistered => "Leader key has already been registered",
+            Error::LeaderKeyBadConsensusHash => "Leader key has an invalid consensus hash",
+
+            Error::UserBurnSupportBadConsensusHash => "User burn support has an invalid consensus hash",
+            Error::UserBurnSupportNoLeaderKey => "User burn support does not match a registered leader key",
         }
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
-pub enum CheckResult {
-    BlockCommitOk,
-    BlockCommitPredatesGenesis,
-    BlockCommitBadEpoch,
-    BlockCommitNoLeaderKey,
-    BlockCommitLeaderKeyAlreadyUsed,
-    BlockCommitNoParent,
-    BlockCommitBadInput,
+#[derive(Debug, PartialEq, Clone, Eq)]
+pub struct LeaderBlockCommitOp {
+    pub block_header_hash: BlockHeaderHash, // hash of Stacks block header (double-sha256)
+    pub new_seed: VRFSeed,                  // new seed for this block
+    pub parent_block_ptr: u32,              // pointer to the block that contains the parent block hash 
+    pub parent_vtxindex: u16,               // offset in the parent block where the parent block hash can be found
+    pub key_block_ptr: u32,                 // pointer to the block that contains the leader key registration 
+    pub key_vtxindex: u16,                  // offset in the block where the leader key can be found
+    pub memo: Vec<u8>,                      // extra unused byte
 
-    LeaderKeyOk,
-    LeaderKeyAlreadyRegistered,
-    LeaderKeyBadConsensusHash,
+    pub burn_fee: u64,                      // how many burn tokens (e.g. satoshis) were destroyed to produce this block
+    pub input: BurnchainSigner,             // burn chain keys that must match the key registration
 
-    UserBurnSupportOk,
-    UserBurnSupportBadConsensusHash,
-    UserBurnSupportNoLeaderKey,
+    // common to all transactions
+    pub txid: Txid,                         // transaction ID
+    pub vtxindex: u32,                      // index in the block where this tx occurs
+    pub block_height: u64,                  // block height at which this tx occurs
+    pub burn_header_hash: BurnchainHeaderHash,      // hash of the burn chain block header
```

Comment:
Keep being confused by the naming + comment:
`hash of the burn chain block header`, or `hash of burn chain block` (same file, line 177) ?

Either way, this field is supposed to be hydrated with the hash of the parent block, not the hash of the block including this op, right?
If so, there might be an issue in `sortition.rs`:
https://github.com/blockstack/blockstack-core/blob/8635433074d30e7567e8578ebf6f5303b08716cf/src/chainstate/burn/sortition.rs#L148-L150

Where I think we should be fetching the `op` using the parent block hash of the block where the sortition is happening, instead of `last_sortition_snapshot.burn_header_hash`.